### PR TITLE
Use ABI3 for `perspective-python` shared library

### PR DESCRIPTION
**Pull request by texodus**
Adds `abi3-py39` build target to `perspective-python`

### CODE REVIEW

File: rust/perspective-python/src/client/client_sync.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ pub struct PySyncClient(PyClient);
 #[pymethods]
 impl PySyncClient {
     #[new]
-    pub fn new(callback: Py<PyFunction>) -> PyResult<Self> {
+    pub fn new(callback: Py<PyAny>) -> PyResult<Self> {
```
Comment (timkpaine) [inline rust/perspective-python/src/client/client_sync.rs:43]: TODO check for call

**Review by timkpaine — CHANGES_REQUESTED** [review]
Tests/any other places should remove spurious lambdas as in https://github.com/finos/perspective/pull/2667

File: rust/perspective-python/src/client/client_sync.rs (hunk 1 above)
Comment (texodus) [reply rust/perspective-python/src/client/client_sync.rs:43]: The previous version of Perspective did not do this, and Python has adequate error handling for this already.

Comment (timkpaine) [reply rust/perspective-python/src/client/client_sync.rs:43]: Its quite bit uglier now though:
```raw
perspective.PerspectivePyError: External error: PyErr { type: <class 'TypeError'>, value: TypeError("'<Type>' object is not callable"), traceback: Some(<traceback object at 0x15c1e5780>) }
```

**Comment by texodus** [comment]
Fixes #2632

File: rust/perspective-python/src/client/client_sync.rs (hunk 1 above)
Comment (texodus) [reply rust/perspective-python/src/client/client_sync.rs:43]: I've fixed this error message on #2659

**Pull request merged**